src/core/src/types/mod.rs

```diff
@@ -18,3 +18,6 @@ pub use self::complex::Complex;
 
 mod arrays;
 pub use self::arrays::{Array2, Array3};
+
+mod thread_local_store;
```

Comment:
I went with a public `lumol::parallel` module. I felt it was going to be needed anyways, tell me if you don't like it.